core/src/meta.rs

```diff
@@ -0,0 +1,130 @@
+use quote::{ToTokens, __private::Span};
+use syn::*;
+
+#[derive(Debug)]
+pub struct Struct {
+    pub name: String,
+    pub methods: Vec<Method>,
+}
+
+#[derive(Debug)]
+pub struct Method {
+    pub name: String,
+    pub full_path_name: String,
+    pub self_param: Option<Param>,
+    pub params: Vec<Param>,
+    pub return_type: Option<Type>,
```

Comment:
the types used here should probably be ids or names or something, which you can use to fetch the actual type from some context object